common/src/main/java/com/wynntils/features/user/overlays/PowderAbilityBarOverlayFeature.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user.overlays;
+
+import com.google.common.collect.ImmutableList;
+import com.mojang.blaze3d.platform.Window;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.overlays.Overlay;
+import com.wynntils.core.features.overlays.OverlayPosition;
+import com.wynntils.core.features.overlays.annotations.OverlayInfo;
+import com.wynntils.core.features.overlays.sizes.GuiScaledOverlaySize;
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.mc.objects.CustomColor;
+import com.wynntils.mc.render.FontRenderer;
+import com.wynntils.mc.render.HorizontalAlignment;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.mc.render.VerticalAlignment;
+import com.wynntils.wynn.model.ActionBarModel;
+import com.wynntils.wynn.objects.Powder;
+import com.wynntils.wynn.utils.WynnUtils;
+
+public class PowderAbilityBarOverlayFeature extends UserFeature {
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    private final Overlay powderAbilityBarOverlay = new PowderAbilityBarOverlay();
+
+    @Override
+    protected void onInit(
+            ImmutableList.Builder<Condition> conditions, ImmutableList.Builder<Class<? extends Model>> dependencies) {
+        dependencies.add(ActionBarModel.class);
+    }
+
+    public static class PowderAbilityBarOverlay extends Overlay {
+        @Config
+        public FontRenderer.TextShadow textShadow = FontRenderer.TextShadow.OUTLINE;
+
+        @Config
+        public boolean flip = false;
+
+        public PowderAbilityBarOverlay() {
+            super(
+                    new OverlayPosition(
+                            -30,
+                            150,
+                            VerticalAlignment.Bottom,
+                            HorizontalAlignment.Center,
+                            OverlayPosition.AnchorSection.BottomMiddle),
+                    new GuiScaledOverlaySize(81, 21));
+        }
+
+        @Override
+        public void render(PoseStack poseStack, float partialTicks, Window window) {
+            float powderSpecialCharge = ActionBarModel.getPowderSpecialCharge();
+            if (!WynnUtils.onServer() || powderSpecialCharge == -1) return;
```

Comment:
The onServer is not needed anymore. The render event will not be sent unless we are on a server.

Reply:
Also, when `powderSpecialCharge == -1`, it might be nice to show empty `powderSpecialCharge` of `0` instead

Reply:
Id say that implies the weapon has or had a known special attack. So a gray bar on has but unknown might better convey the current state. Id also like an option to hide the bar when no weapon is held